test/integration/clustermanager_test.go

```diff
@@ -267,4 +267,28 @@ var _ = ginkgo.Describe("ClusterManager", func() {
 			return true
 		}, eventuallyTimeout, eventuallyInterval).Should(gomega.BeTrue())
 	})
+
+	ginkgo.It("Cluster manager should have correct degraded conditions", func() {
```

Comment:
I don't understand how we can expect the deployment to exist, since there is no `BeforeSuite` creating a `ClusterManager`, but I feel I must be missing something about ginkgo here. I don't want to be pedantic and force you to use a `BeforeEach` if it's not required, will ping you later on slack about this so we can chat about it.

Reply:
OK, Indeed current test case is a bit confused，the test cases
https://github.com/open-cluster-management/registration-operator/blob/master/test/integration/clustermanager_test.go#L172
https://github.com/open-cluster-management/registration-operator/blob/master/test/integration/clustermanager_test.go#L227 and 
new degrade test case are depend on https://github.com/open-cluster-management/registration-operator/blob/master/test/integration/clustermanager_test.go#L76 (it will create a `ClusterManager`)

the `ClusterManager` has only one fixed namespace in hub cluster (`open-cluster-management-hub`), and controller runtime envtest does not enable the garbage collector, so I think we cannot create `ClusterManager` on `BeforeEach` and delete the `ClusterManager` on `AfterEach`, this because when we delete `ClusterManager`, the cluster manager controller will delete the `ClusterManager` namespace, because there is no garbage collector, the namespace will keep the terminating status

I try to refactor the test cases
1. put the create `ClusterManager` action on `BeforeSuite`
2. use two context to organize the test cases

I think this will be more clear than before, I will ping you on slack